tmt/steps/execute/__init__.py

```diff
@@ -854,7 +786,20 @@ def extract_tmt_report_results(self, invocation: TestInvocation) -> list["tmt.Re
 
         collection.validate()
 
-        return self._process_results_reduce(invocation, collection.results)
+        # TODO: Can we use the Result.from_invocation(...) here? Should I somehow map the results
+        # to the test invocation here or not?
+        results = [tmt.Result.from_serialized(r) for r in collection.results]
+
+        # TODO: Should I somehow map the Result to test invocation?
+        for result in results:
+            # Ensure the result name is in test namespace
+            # TODO: Do we want to move it under main test namespace or not, I suppose we do.
```

Comment:
I suppose we want to add a parent test name as a namespace/prefix to all subresults, correct?

Reply:
I've moved the namespacing into the `to_subresult(invocation)` method of the `tmt.Result`. If think it makes sense to have it there. I still have a feeling the namespacing should be probably enforced if the Result **gets converted** to a SubResult.

The framework can now pass the results unmodified (it will not call the `to_subresult` method).

- Or we can modify the behavior of the `to_subresult` method with some flags, e.g. `to_subresult(add_test_namespace=False)`.
- Or we can optionally add namespace only if the `invocation` argument of `to_subresult(invocation)` is set.

Reply:
Marking as resolved. We decided (in the comments below) to remove the main test prefix and keep the subresult name on the test framework.